### DISCUSSION TITLE
classes/class_concavepolygonshape2d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#97] [User-contributed notes]
# classes/class_concavepolygonshape2d

Inherits: Shape2D< Resource< RefCounted< Object A 2D polyline shape used for physics collision. Description: A 2D polyline shape, intended for use in physics. Used internally in CollisionPolygon2D ...

https://docs.godotengine.org/en/stable/classes/class_concavepolygonshape2d.html

**Comment by vbettaque** [comment]
The name of this resource is highly misleading. It describes a general polyline, whose segments do not even have to be connected to each other. Only when they do and the ends join does it actually describe a (concave) polygon.

**Reply by Calinou** [reply]
It's the 2D analog to ConcavePolygonShape3D (also called a "trimesh" collision shape). It's called this way because it covers use cases that ConvexPolygonShape2D cannot cover by design without using multiple shapes.

ConcavePolygonShape3D can also have disjointed, non-manifold vertices.
